src/Documentation/Advanced-Concepts/Activation-Garbage-Collection.md

```diff
@@ -36,12 +27,73 @@ The most common approach is to configure the silo to periodically scan for activ
 
 **Collection Age Limit**
 
-Activations that haven't been used within a period of time are deactivated automatically by the runtime. This period of time is called the collection age limit.
+This period of time after which an idle grain activation becomes subject to Activation GC is called Collection Age Limit. The default Collection Age Limit is 2 hours, but it can be changed globally or for individual grain classes.
+
+## Explicit Control of Activation Garbage Collection
+
+### Delaying Activation GC
+
+A grain activation can delay its own Activation GC, by calling `this.DelayDeactivation()` method:
+
+``` csharp
+protected void DelayDeactivation(TimeSpan timeSpan)
+```
+
+This call will ensure that this activation is not deactivated for at least the specified time duration. It takes priority over Activation Garbage Collection settings specified in the config, but does not cancel them.
+Therefore, this call provides an additional hook to **delay the deactivation beyond what is specified in the Activation Garbage Collection settings**. This call can not be used to expedite Activation Garbage Collection.
+
+
+A positive <c>`timeSpan`</c> value means “prevent GC of this activation for that time span”.
+
+A negative <c>`timeSpan`</c> value means “cancel the previous setting of the `DelayDeactivation` call and make this activation behave based on the regular Activation Garbage Collection settings”.
+
+**Scenarios:**
+
+1) Activation Garbage Collection settings specify age limit of 10 minutes and the grain is making a call to DelayDeactivation(20 min), it will cause this activatin to not be collected for at least 20 min.
+
+2) Activation Garbage Collection settings specify age limit of 10 minutes and the grain is making a call to DelayDeactivation(5 min), the activation will be collected after 10 min, if no extra calls were made.
+
+3) Activation Garbage Collection settings specify age limit of 10 minutes and the grain is making a call to DelayDeactivation(5 min), and after 7 minutes there is another call on this grain, the activation will be collected after 17 min from time zero, if no extra calls were made.
+
+4) Activation Garbage Collection settings specify age limit of 10 minutes and the grain is making a call to DelayDeactivation(20 min), and after 7 minutes there is another call on this grain, the activation will be collected after 20 min from time zero, if no extra calls were made.
 
- The runtime's default collection age limit is 2 hours.
+Note that `DelayDeactivation` does not 100% guarantee that the grain activation will not get deactivated before the specified period of time expires. There are certain failure cases that may cause 'premature' deactivation of grains.
```

Comment:
this note is something VERY important that people sometime miss, and think they can pin a grain activation to a silo forever. Maybe you want to be a little bit more explicit about what are the failure scenarios, or at least highlight that delaying deactivation should not be used to make an activation exist forever during the lifetime of the silo it lives in.

Reply:
Good idea. I added a couple of sentences to stress that.